### PR TITLE
Setup knative

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: nix/modules/k3s.nix [hunk 1]
```diff
@@ -45,36 +36,35 @@ in
...
+      #80
+      ## istio https port
+      #443
+    ];
 
     virtualisation.containerd.enable = true;
 
     virtualisation.containerd.settings = {
-      # Upstream this?
-      plugins.cri.containerd.snapshotter = lib.mkIf hasZfs "zfs";
```
Comment (Mic92) [inline nix/modules/k3s.nix:58]: I will merge this one soon: https://github.com/NixOS/nixpkgs/pull/125169

File: nix/modules/vhive.nix [hunk 2]
```diff
@@ -3,6 +3,7 @@
   systemd.services.vhive = {
     wantedBy = ["multi-user.target"];
     serviceConfig = {
+      path = [ pkgs.nettools pkgs.kubectl pkgs.iptables ];
```
Comment (Mic92) [inline nix/modules/vhive.nix]: This should be one level up.

File: nix/modules/k3s.nix (hunk 1 above)
Comment (Mic92) [reply nix/modules/k3s.nix:58]: Done.

File: nix/modules/vhive.nix [hunk 3]
```diff
@@ -1,15 +1,21 @@
 { pkgs, ... }:
 {
-  systemd.services.vhive = {
+  systemd.services.vhive = let
+    preStart = ''
+      rm -rf /etc/firecracker-containerd/fccd-cri.sock
+      ip --json link | jq -r '.[] | select(.ifname | test(".*_tap$|br0|br1")) | .ifname' | xargs -r -n1 ip link del
```
Comment (pogoba) [inline nix/modules/vhive.nix:6]: +reputation

**Review by pogoba — APPROVED** [review]

**Pull request merged**